src/warc2zim/url_rewriting.py

```diff
@@ -71,7 +71,7 @@
         r"=[^&]+).*",
         "replace": r"youtube.fuzzy.replayweb.page/\1\2",
     },
-    {"pattern": r"([^?]+\?)[\d]+$", "replace": r"\1"},
+    {"pattern": r"([^?]+)\?[\d]+$", "replace": r"\1"},
```

Comment:
No, this is a change I decided to make, see explanation in first comment paragraph in "A change in the fuzzy rule which removes "digits-only" query parameter". I'm not 100% sure it is a good thing, but handling this exception (a URL with a trailing `?` and no querystring) is going to be dirty otherwise.

Reply:
OK then this definitely mandates a comment in the code.
I don't understand exactly the value of the change TBH. What's the problem if we keep it?

Reply:
If we keep it, since browsers do not send a trailing `?` when present in a URL, WARC records won't be present with trailing `?` as target URI. So we will need to do some custom matching saying "if the URL we are searching for has a trailing `?`, search for a WARC record without it". I feel like this is a bit hackhish

Reply:
OK, let's get @mgautierfr's opinion on this then

Reply:
> Maybe this is why I was needing to add the extra rules for JS only working on already rewritten url (to check)

I'm not sure to understand what you are referring too, could you explain?

Reply:
I'm speaking about this rule : https://github.com/openzim/warc2zim/blob/warc2zim2/src/warc2zim/url_rewriting.py#L84-L90

It appears that youtube player at a moment works on transformed url and add a query string to it. We have to remove it. This was not necessary in warc2zim1 as the inserted wombat was not doing fuzzy matching transformation (and so, querystring was removed using fuzzy rules (in wabac) matching original url)

My point is that **maybe**, the youtube player adding a query string is because of this trailing `?`. But as said, I haven't fully search the root cause, so it may be unrelated.

Reply:
I just did a test without this extra rule and I can confirm it is working perfectly fine. I propose to remove them  for now, it will be easy to add them back should we realize there is a problem again.